x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -722,6 +736,8 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
                           showBulkDuplicateExceptionsConfirmation={showBulkDuplicateConfirmation}
                           showManualRuleRunConfirmation={showManualRuleRunConfirmation}
                           confirmDeletion={confirmDeletion}
+                          isRevertBaseVersionDisabled={isRevertBaseVersionDisabled}
+                          openRuleDiffFlyout={openFlyout}
```

Comment:
It looks like inversion of control is required for `RuleActionsOverflow`. Increasing the number of component props has a limited capacity. In general it reduces composability, readability and maintainability. In that case we have `isRevertBaseVersionDisabled` and `openFlyout` leaking through. Though it might be simpler to use the flyout management hook under the hood.

Anyway the refactoring might be quite bulky and should happen outside of this PR.